plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcModule.java

```diff
@@ -50,7 +50,6 @@ public void setup(Binder binder)
         install(new JdbcDiagnosticModule());
         install(new IdentifierMappingModule());
         install(new RemoteQueryModifierModule());
-        install(new SyntheticColumnHandleBuilderModule());
```

Comment:
The general question here is whether we use DI properly in the project or we don't. I'm not sure whether adding a 'new' in the production code is a move towards a good direction. I remember we had this discussion on the other review, that if in the future this SyntheticColumnHandleBuilder starts having dependencies, then updating all the places where 'new' is used to build it, will increase the maintenance cost.

Reply:
Yes but with this change I'm confident that the class is stateless (the only parameter we need is the length which needs to be passed per call anyway).

If the need arises in future we can introduce Guice back.

(we don't have RULES, we try to do what makes sense where - in this case for current usages DI appears overkill).

Reply:
To me it looks like you are trying to make an 'inverted' justification compared to the flag one described under 'motivation' in Guice user guide: https://github.com/google/guice/wiki/Motivation
where there is exactly that 'anti-pattern' with factories shown and they refactor to show how you can do it with DI.

Here we have for instance JdbcMetadataFactory that has the 'new' calls and that prevents us from using DI properly here and this is why you feel that adding a module is an overkill.

What I wanted to say is that it would look consistent if either:
- [A] we did not use DI and used only plain java with factories OR
- [B] used DI as described in Guice manual
It is neither one nor the other and that is why we still keep on NOT closing conversations around this piece of code and that is exactly why I'm making one commit, and on the next commit you are reversing my changes. Looks like threading water to me. I hope we can find something agreeable/reasonable and stick with it. If you have a preference to have [A] or [B], then let me know. For me an answer saying, we can choose [A] or [B] depending on the weather feels unprofessional, so I'd like to avoid it if possible.

Reply:
Sorry, maybe I was not totally transparent of what I'm thinking about. I linked the 'motivation' document that is not applicable here directly. I have read the code and understood that what I should have pointed you at is guice's assisted inject: https://github.com/google/guice/wiki/AssistedInject

For your convenience I have provided a sample code (like we agreed on the meeting) of how can the boilerplate code for factories be avoided:

b3428a76f48881f6d18d4a09b72e66315fd4b0c1

This is also in this draft PR: https://github.com/trinodb/trino/pull/19043/ if it's easier to review the changes that way.

What I wanted to make you aware of, is that if we don't mix layers of Guice | no-Guice | Guice, then the code becomes more manageable (if the factories boilerplate is removed it is not that annoying to add another dependent class like Synthetic..Builder).

ps. In the example commit I wasn't aiming for making the build green and reworking the test, as I wanted to cut corners on this 'preview' commit. I still don't know whether it will be something that you want to pursue.

Reply:
no longer applicable since we don't have this class anymore